src/EFCore.PG.NodaTime/Query/Internal/NpgsqlNodaTimeMethodCallTranslatorPlugin.cs

```diff
@@ -0,0 +1,184 @@
+        private static readonly MethodInfo Period_FromMonths  = typeof(Period).GetRuntimeMethod(nameof(Period.FromMonths),       new[] { typeof(int) })!;
+        private static readonly MethodInfo Period_FromWeeks   = typeof(Period).GetRuntimeMethod(nameof(Period.FromWeeks),        new[] { typeof(int) })!;
+        private static readonly MethodInfo Period_FromDays    = typeof(Period).GetRuntimeMethod(nameof(Period.FromDays),         new[] { typeof(int) })!;
+        private static readonly MethodInfo Period_FromHours   = typeof(Period).GetRuntimeMethod(nameof(Period.FromHours),        new[] { typeof(long) })!;
+        private static readonly MethodInfo Period_FromMinutes = typeof(Period).GetRuntimeMethod(nameof(Period.FromMinutes),      new[] { typeof(long) })!;
+        private static readonly MethodInfo Period_FromSeconds = typeof(Period).GetRuntimeMethod(nameof(Period.FromSeconds),      new[] { typeof(long) })!;
+
+        /// <summary>
+        /// The mapping of supported method translations.
+        /// </summary>
+        private static readonly Dictionary<MethodInfo, string> PeriodMethodMap = new()
+        {
+            { typeof(Period).GetRuntimeMethod(nameof(Period.FromYears),        new[] { typeof(int) })!,  "years" },
+            { typeof(Period).GetRuntimeMethod(nameof(Period.FromMonths),       new[] { typeof(int) })!,  "months" },
+            { typeof(Period).GetRuntimeMethod(nameof(Period.FromWeeks),        new[] { typeof(int) })!,  "weeks" },
+            { typeof(Period).GetRuntimeMethod(nameof(Period.FromDays),         new[] { typeof(int) })!,  "days" },
+            { typeof(Period).GetRuntimeMethod(nameof(Period.FromHours),        new[] { typeof(long) })!, "hours" },
+            { typeof(Period).GetRuntimeMethod(nameof(Period.FromMinutes),      new[] { typeof(long) })!, "mins" },
+            { typeof(Period).GetRuntimeMethod(nameof(Period.FromSeconds),      new[] { typeof(long) })!, "secs" },
+        };
+
+        private static readonly bool[][] TrueArrays =
+        {
+            Array.Empty<bool>(),
+            new[] { true },
+            new[] { true, true },
+        };
+
+        public NpgsqlNodaTimeMethodCallTranslator(
+            IRelationalTypeMappingSource typeMappingSource,
+            NpgsqlSqlExpressionFactory sqlExpressionFactory)
+        {
+            _typeMappingSource = typeMappingSource;
+            _sqlExpressionFactory = sqlExpressionFactory;
+        }
+
+#pragma warning disable EF1001
+        /// <inheritdoc />
+        public virtual SqlExpression? Translate(
+            SqlExpression? instance,
+            MethodInfo method,
+            IReadOnlyList<SqlExpression> arguments,
+            IDiagnosticsLogger<DbLoggerCategory.Query> logger)
+        {
+            if (method == SystemClock_GetCurrentInstant)
+            {
+                return NpgsqlNodaTimeTypeMappingSourcePlugin.LegacyTimestampBehavior
+                    ? _sqlExpressionFactory.AtTimeZone(
```

Comment:
Not exactly... [The old (and now legacy) translation for GetCurrentInstant](https://github.com/npgsql/efcore.pg/blob/v5.0.10/src/EFCore.PG.NodaTime/Query/ExpressionTranslators/Internal/NpgsqlNodaTimeMethodCallTranslatorPlugin.cs#L84) was `NOW() AT TIME ZONE 'UTC'` (with constant UTC), which returns a UTC timestamp but typed as `timestamp without time zone`, since that is how we were mapping Instant. So no timezone conversion - just a PG type conversion.